moshi-kotlin-tests/src/test/kotlin/com/squareup/moshi/kotlin/DualKotlinTest.kt

```diff
@@ -243,6 +243,8 @@ class DualKotlinTest {
     assertThat(decoded.a).isEqualTo(null)
   }
 
+  // TODO CR ValueClass enables generateAdapter but it's not present in reflect-only.
```

Comment:
I'm not actually sure why this didn't fail before tbh. We previously allowed reflect to read these if they weren't found so people could have shared source that selectively enabled code gen. Open to either approach, just need to make a call here